app/services/hyrax/find_objects_via_solr_service.rb

```diff
@@ -14,13 +16,15 @@ class << self
       # @param join_with [String] the value we're joining the clauses with (default: ' OR ' for backward compatibility with ActiveFedora where)
       # @param type [String] The type of query to run. Either 'raw' or 'field' (default: 'field')
       # @param use_valkyrie [Boolean] if true, return Valkyrie resource(s); otherwise, return ActiveFedora object(s)
-      # @return [Array<ActiveFedora|Valkyrie::Resource>] objects matching the query
+      # @return [Array<ActiveFedora::Base|Valkyrie::Resource>] objects matching the query
       def find_for_model_by_field_pairs(model:, field_pairs:, join_with: ' OR ', type: 'field', use_valkyrie: Hyrax.config.use_valkyrie?)
-        return model.where(field_pairs).to_a unless use_valkyrie
-        query = solr_query_builder.construct_query_for_model(model, field_pairs, join_with, type)
-        results = solr_service.query(query)
-        ids = results.map(&:id)
-        query_service.find_many_by_ids(ids: ids).to_a
+        Deprecation.warn("'##{__method__}' will be removed in Hyrax 4.0.  " \
+                         "Instead, use 'Hyrax::SolrQueryService.new.with_model(...).with_field_pairs(...).get_objects'.")
+
+        solr_query_builder.new
+                          .with_model(model: model)
+                          .with_field_pairs(field_pairs: field_pairs, join_with: join_with, type: type)
+                          .get_objects(use_valkyrie: use_valkyrie).to_a
```

Comment:
FindObjectViaSolrService #find_for_model_by_field_pairs needs to continue to return an Array for backward compatibility.

This method is no longer used in any Hyrax code.